core/server/services/members/middleware.js

```diff
@@ -0,0 +1,71 @@
+const common = require('../../lib/common');
+const shared = require('../../web/shared');
+const labsService = require('../labs');
+const membersService = require('./index');
+
+module.exports.use = (siteApp) => {
+    // @TODO only loads this stuff if members is enabled
+    // Set req.member & res.locals.member if a cookie is set
+    siteApp.get('/members/ssr', shared.middlewares.labs.members, async function (req, res) {
```

Comment:
IMO the members service shouldn't be in charge of where this middleware is mounted.

Say for example we were to have many services doing this approach, it would be really easy to have collisions  - it seems to me that site/app main responsibility is matching routes to functionality - what do you think?

Reply:
@allouis So do you think a better approach would be:
(1) passing into this `use` method an endpoint it should mount on, like `/members/` for endpoints and `/public/` for the static file
or
(2) should we rather expose each  handler as a method and map them individually in the `site/app.js` like: 
```js
// for static files
siteApp.use('/public/members.js'membersMiddleware.public); // this doesn't make much sense tbh :man_shrugging: 

// for routes and req/res decoration
siteApp.get('/members/ssr', membersMiddleware.login);
siteApp.delete('/members/ssr', membersMiddleware.logout);
siteApp.use(membersMiddleware)
```

Don't think either method is ideal, but 1st one is a lot less code on the initialization site and still prevents the collision I think.

Reply:
IMO (2) is a nice solution? It is really explicit as to what's happening, and if you wanted didn't know, you could guess which file to look in to find where things happen?

Reply:
Cool. I think (2) is a little too explicit but we can combine it into something later if we find this pattern bad :) Still, a lot cleaner then what is there now!

Reply:
Will roll with it. Thanks!

Reply:
> can combine it into something later if we find this pattern bad

Yeah good call :relaxed: